### ISSUE TITLE
Preprocess Bytecodes to remove uses of Vector, Hashtable, and Stack in JS Port

### ISSUE THREAD

**Issue by shannah** [codenameone/CodenameOne#2073]
The presence of legacy collection classes Vector, Hashtable, and Stack cause problems for TeaVM because it forces it to treat all collection classes as potentially synchronized.  This might affect performance, and it's a nuisance when working on code that must run in the main javascript thread.

Since most use-cases of these classes are just legacy (don't rely on synchronization), I'd like to create an ASM preprocessor that will "de-synchronize" all of these legacy classes before running through TeaVM.  It shouldn't be hard to do.  Just implement alternative implementations that aren't synchronized, and change all references to Vector, Hashtable, and Stack to their corresponding non-synchronized implementations.

This could optionally be turned off with a build hint.

This doesn't stop someone from still implementing `List` with methods that are async.. this will still cause TeaVM to treat all Lists as async.. But it will cover a lot of the use cases.

Reference TeaVM issue https://github.com/konsoletyper/teavm/issues/248

**Comment by codenameone** [comment]
Why not just edit the TeaVM implementation of these classes and remove the synchronized statements?

**Comment by shannah** [comment]
Yeah.  That would probably be easier.  Alexey accepted my pull request to add add an annotation that suppresses the "sync errors", which mostly resolves the issue with being a nuisance when running on the main thread.  So the problem now is just performance, and is only theoretical... I don't have hard numbers to show how much "slower" it is with Vector.   I'll have to get those numbers before proceeding.

**Comment by codenameone** [comment]
wouldn't it just be a replace all for `public synchronized` to `public` and the same for private and protected?

Seems like synchronized is just a waste on the JS port and this seems like a 4 minute fix.

**Comment by shannah** [comment]
Yeah.  Quick fix.  But recompiling TeaVM take 5 minutes.  Testing the change takes 5 to 10 minutes.  And pushing out a new version of TeaVM to the maven repo is about 30 minutes.

I'll wait until I have another change ready to roll and I'll include it with that.

**Comment by shai-almog** [comment]
Closing per the issue-tracker triage (see [`NEEDS_REVIEW.md`](https://github.com/codenameone/CodenameOne/blob/master/NEEDS_REVIEW.md)).

Referred to the old TeaVM-based JS port. The current [`Ports/JavaScriptPort/`](https://github.com/codenameone/CodenameOne/tree/master/Ports/JavaScriptPort) is a different (Heavy*Impl-based) port; the TeaVM constraint described no longer applies.
